topology/probes/k8s/probe.go

```diff
@@ -74,37 +72,32 @@ func (p *Probe) Stop() {
 	}
 }
 
-// NewProbe creates the probe for tracking k8s events
-func NewProbe(manager, clusterName string, g *graph.Graph, subprobes map[string]Subprobe, linkers []probe.Probe) (*Probe, error) {
-	clusterNode := g.NewNode(graph.GenID(), NewMetadata(manager, "cluster", nil, clusterName), "")
-	for _, subprobe := range subprobes {
-		if cache, ok := subprobe.(*ResourceCache); ok {
-			if cache.handler.IsTopLevel() {
-				if clusterLinker := newClusterLinker(g, clusterNode, cache); clusterLinker != nil {
-					linkers = append(linkers, clusterLinker)
-				}
-			}
+// AppendClusterLinkers appends newly created cluster linker per type
+func (p *Probe) AppendClusterLinkers(types ...string) {
+	clusterIndexer := newClusterIndexer(p.graph)
+	for _, ty := range types {
+		if clusterLinker := newClusterLinker(p.graph, clusterIndexer, p.subprobes, p.manager, ty); clusterLinker != nil {
+			p.linkers = append(p.linkers, clusterLinker)
 		}
 	}
-
-	return &Probe{
-		subprobes: subprobes,
-		linkers:   linkers,
-	}, nil
-}
-
-type clusterLinker struct {
-	graph.DefaultLinker
-	graph       *graph.Graph
-	clusterNode *graph.Node
 }
 
-func (cl *clusterLinker) GetBALinks(node *graph.Node) []*graph.Edge {
-	id := graph.GenID(string(cl.clusterNode.ID), string(node.ID), "RelationType", topology.OwnershipLink)
-	return []*graph.Edge{cl.graph.CreateEdge(id, cl.clusterNode, node, topology.OwnershipMetadata(), time.Now(), "")}
+// AppendNamespaceLinkers appends newly created namespace linker per type
+func (p *Probe) AppendNamespaceLinkers(types ...string) {
+	namespaceIndexer := newNamespaceIndexer(p.graph)
+	for _, ty := range types {
+		if namespaceLinker := newNamespaceLinker(p.graph, namespaceIndexer, p.subprobes, p.manager, ty); namespaceLinker != nil {
```

Comment:
I would rather create a filter as it was previously based on the array of the types using it to create the single indexer. So that is not statically defined and we keep only one indexer.